interfaces/builtin/socketcan_control.go

```diff
@@ -0,0 +1,64 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const socketCanControlSummary = `allows configuration and use of SocketCAN network interfaces`
+
+const socketCanControlBaseDeclarationSlots = `
+  socketcan-control:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const socketCanControlConnectedPlugAppArmor = `
+# Description: Can configure and use SocketCAN networking
+network can,
+
+# Allow configuration of the interface using the ip command for SocketCAN
+/{,usr/}{,s}bin/ip ixr,
+
+# required by ip to configure SocketCAN networks
+capability net_admin,
+network netlink raw,
+`
+
+const socketCanControlConnectedPlugSecComp = `
+# Description: Can use SocketCAN networking
+bind
+
+# required by ip to configure SocketCAN networks
+socket AF_NETLINK - NETLINK_ROUTE
+socket AF_NETLINK - NETLINK_GENERIC
```

Comment:
Similarly, these are not specific to CAN.

The split here seems a bit suspect. It seems to grant the same thing that the use of can, above, plus the configuration of the network in general.

Can we please have a call tomorrow? I may be completely off, but would like to at least understand this a bit better if you could spare a moment.

Reply:
The reasoning behind both of this is captured in these comments:
* https://github.com/snapcore/snapd/pull/5279#issuecomment-395735128
* https://github.com/snapcore/snapd/pull/5279#issuecomment-395738522

Basically, yes, there is overlap between can-bus/network and can-bus-control/network-control (net_admin is in a lot of ways a catch all, like sys_admin-- the Linux capabilities are unfortunately not finegrained). However, network-control is *big* and we felt that something that only wanted to administer can-bus didn't need all of network-control. Specifically, it doesn't need to change hostname, bridges, inet, inet6, firewalling, ieee80211, rfkill, resolvconf, tun/tap, network namespaces, sys_admin, etc. It only needs 'ip', net_admin, and AF_CAN. These NETLINK_ROUTE and NETLINK_GENERIC are similarly not CAN-specific, but required by the ip command. Put another way, we broke it out since CAN devices might play a role in human safety and limiting what both can-bus and can-bus-control interfaces could do was deemed a good thing.